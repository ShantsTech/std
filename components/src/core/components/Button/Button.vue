<!--
/*
 * This file is part of Shants Tech LLC
 *
 * Copyright (C) 2020 Shants Tech LLC
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */
-->

<template>
  <button type="button" :class="classes" :style="style">
    <slot name="icon">
      <std-icon v-if="iconName" :name="iconName" class="std-button-icon" />
    </slot>
    {{ label }}
    <slot name="iconRight">
      <std-icon
        v-if="iconRightName"
        :name="iconRightName"
        class="std-button-icon"
      />
    </slot>
  </button>
</template>

<script lang="ts">
import {defineComponent} from 'vue';
import {
  ButtonSize,
  ButtonType,
  SIZES,
  SIZE_MEDIUM,
  TYPES,
  TYPE_MAIN,
} from './types';
import Icon from '@ohrm/std/core/components/Icon/Icon.vue';

export default defineComponent({
  name: 'std-button',

  components: {
    'std-icon': Icon,
  },

  props: {
    label: {
      type: String,
      required: true,
    },
    displayType: {
      type: String,
      default: TYPE_MAIN,
      validator: function(value: ButtonType) {
        return TYPES.indexOf(value) !== -1;
      },
    },
    size: {
      type: String,
      default: SIZE_MEDIUM,
      validator: function(value: ButtonSize) {
        return SIZES.indexOf(value) !== -1;
      },
    },
    style: {
      type: Object,
    },
    iconName: {
      type: String,
    },
    iconRightName: {
      type: String,
    },
  },

  computed: {
    classes(): object {
      return {
        'std-button': true,
        [`std-button--${this.size}`]: true,
        [`std-button--${this.displayType}`]: true,
      };
    },
  },
});
</script>

<style src="./button.scss" lang="scss" scoped></style>
