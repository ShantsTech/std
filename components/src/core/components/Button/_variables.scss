// This file is part of Shants Tech LLC
//
// Copyright (C) 2020 Shants Tech LLC
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

$std-button-min-width: 100px !default;
$std-button-horizontal-padding: 32px !default;
$std-button-vertical-padding--large: 16px !default;
$std-button-vertical-padding--medium: 8px !default;
$std-button-vertical-padding--small: 4px !default;
$std-button-font-size--large: 14px !default;
$std-button-font-size--medium: 14px !default;
$std-button-font-size--small: 12px !default;
$std-button-font-weight: 600 !default;
$std-button-font-family: $std-font-family !default;
$std-button-font-color: $std-white-color !default;

$std-list-button-pallete: (
  default: $std-interface-gray-darken-1-color,
  default-darken-5: $std-interface-gray-darken-5-color,
  default-lighten-5: $std-interface-gray-lighten-5-color,
  default-alpha-10: $std-interface-gray-alpha-10-color,
  default-alpha-15: $std-interface-gray-alpha-15-color,
  default-alpha-20: $std-interface-gray-alpha-20-color,
  main: $std-primary-one-color,
  main-darken-5: $std-primary-one-darken-5-color,
  main-lighten-5: $std-primary-one-lighten-5-color,
  main-alpha-10: $std-primary-one-alpha-10-color,
  main-alpha-15: $std-primary-one-alpha-15-color,
  main-alpha-20: $std-primary-one-alpha-20-color,
  secondary: $std-secondary-four-color,
  secondary-darken-5: $std-secondary-four-darken-5-color,
  secondary-lighten-5: $std-secondary-four-lighten-5-color,
  secondary-alpha-10: $std-secondary-four-alpha-10-color,
  secondary-alpha-15: $std-secondary-four-alpha-15-color,
  secondary-alpha-20: $std-secondary-four-alpha-20-color,
  info: $std-feedback-info-color,
  info-darken-5: $std-feedback-info-darken-5-color,
  info-lighten-5: $std-feedback-info-lighten-5-color,
  info-alpha-10: $std-feedback-info-alpha-10-color,
  info-alpha-15: $std-feedback-info-alpha-15-color,
  info-alpha-20: $std-feedback-info-alpha-20-color,
  danger: $std-feedback-danger-color,
  danger-darken-5: $std-feedback-danger-darken-5-color,
  danger-lighten-5: $std-feedback-danger-lighten-5-color,
  danger-alpha-10: $std-feedback-danger-alpha-10-color,
  danger-alpha-15: $std-feedback-danger-alpha-15-color,
  danger-alpha-20: $std-feedback-danger-alpha-20-color,
  warn: $std-feedback-warn-color,
  warn-darken-5: $std-feedback-warn-darken-5-color,
  warn-lighten-5: $std-feedback-warn-lighten-5-color,
  warn-alpha-10: $std-feedback-warn-alpha-10-color,
  warn-alpha-15: $std-feedback-warn-alpha-15-color,
  warn-alpha-20: $std-feedback-warn-alpha-20-color,
  success: $std-feedback-success-color,
  success-darken-5: $std-feedback-success-darken-5-color,
  success-lighten-5: $std-feedback-success-lighten-5-color,
  success-alpha-10: $std-feedback-success-alpha-10-color,
  success-alpha-15: $std-feedback-success-alpha-15-color,
  success-alpha-20: $std-feedback-success-alpha-20-color,
);

/**
 * Solid Buttons
**/

/* Main */
$std-button-color--main: $std-primary-one-color !default;
$std-button-color--main-pressed: map-get(
  $std-list-button-pallete,
  'main-darken-5'
) !default;
$std-button-color--main-hover: map-get(
  $std-list-button-pallete,
  'main-lighten-5'
) !default;
$std-button-color--main-focus: $std-button-color--main-hover !default;
$std-button-color--main-inactive: $std-interface-gray-lighten-2-color !default;

/* Secondary */
$std-button-color--secondary: $std-secondary-four-color !default;
$std-button-color--secondary-pressed: map-get(
  $std-list-button-pallete,
  'secondary-darken-5'
) !default;
$std-button-color--secondary-hover: map-get(
  $std-list-button-pallete,
  'secondary-lighten-5'
) !default;
$std-button-color--secondary-focus: $std-button-color--secondary-hover !default;
$std-button-color--secondary-inactive: $std-interface-gray-lighten-2-color !default;

/* Info */
$std-button-color--info: $std-feedback-info-color !default;
$std-button-color--info-pressed: map-get(
  $std-list-button-pallete,
  'info-darken-5'
) !default;
$std-button-color--info-hover: map-get(
  $std-list-button-pallete,
  'info-lighten-5'
) !default;
$std-button-color--info-focus: $std-button-color--info-hover !default;
$std-button-color--info-inactive: $std-interface-gray-lighten-2-color !default;

/* Danger */
$std-button-color--danger: $std-feedback-danger-color !default;
$std-button-color--danger-pressed: map-get(
  $std-list-button-pallete,
  'danger-darken-5'
) !default;
$std-button-color--danger-hover: map-get(
  $std-list-button-pallete,
  'danger-lighten-5'
) !default;
$std-button-color--danger-focus: $std-button-color--danger-hover !default;
$std-button-color--danger-inactive: $std-interface-gray-lighten-2-color !default;

/* Warning */
$std-button-color--warn: $std-feedback-warn-color !default;
$std-button-color--warn-pressed: map-get(
  $std-list-button-pallete,
  'warn-darken-5'
) !default;
$std-button-color--warn-hover: map-get(
  $std-list-button-pallete,
  'warn-lighten-5'
) !default;
$std-button-color--warn-focus: $std-button-color--warn-hover !default;
$std-button-color--warn-inactive: $std-interface-gray-lighten-2-color !default;

/* Success */
$std-button-color--success: $std-feedback-success-color !default;
$std-button-color--success-pressed: map-get(
  $std-list-button-pallete,
  'success-darken-5'
) !default;
$std-button-color--success-hover: map-get(
  $std-list-button-pallete,
  'success-lighten-5'
) !default;
$std-button-color--success-focus: $std-button-color--success-hover !default;
$std-button-color--success-inactive: $std-interface-gray-lighten-2-color !default;

$std-solid-buttons: (
  main: (
    font: $std-primary-font-color,
    active: $std-button-color--main,
    pressed: $std-button-color--main-pressed,
    focus: $std-button-color--main-focus,
    hover: $std-button-color--main-hover,
    inactive: $std-button-color--main-inactive,
  ),
  secondary: (
    font: $std-secondary-font-color,
    active: $std-button-color--secondary,
    pressed: $std-button-color--secondary-pressed,
    focus: $std-button-color--secondary-focus,
    hover: $std-button-color--secondary-hover,
    inactive: $std-button-color--secondary-inactive,
  ),
  info: (
    font: $std-button-font-color,
    active: $std-button-color--info,
    pressed: $std-button-color--info-pressed,
    focus: $std-button-color--info-focus,
    hover: $std-button-color--info-hover,
    inactive: $std-button-color--info-inactive,
  ),
  danger: (
    font: $std-button-font-color,
    active: $std-button-color--danger,
    pressed: $std-button-color--danger-pressed,
    focus: $std-button-color--danger-focus,
    hover: $std-button-color--danger-hover,
    inactive: $std-button-color--danger-inactive,
  ),
  warn: (
    font: $std-button-font-color,
    active: $std-button-color--warn,
    pressed: $std-button-color--warn-pressed,
    focus: $std-button-color--warn-focus,
    hover: $std-button-color--warn-hover,
    inactive: $std-button-color--warn-inactive,
  ),
  success: (
    font: $std-button-font-color,
    active: $std-button-color--success,
    pressed: $std-button-color--success-pressed,
    focus: $std-button-color--success-focus,
    hover: $std-button-color--success-hover,
    inactive: $std-button-color--success-inactive,
  ),
);

/**
 * Ghost Button
**/
$std-button-color--ghost: $std-secondary-four-color !default;
$std-button-border--ghost: $std-border !default;
$std-button-color--ghost-pressed: map-get(
  $std-list-button-pallete,
  'secondary-darken-5'
) !default;
$std-button-color--ghost-hover: map-get(
  $std-list-button-pallete,
  'secondary-lighten-5'
) !default;
$std-button-color--ghost-focus: $std-button-color--ghost-hover !default;
$std-button-color--ghost-inactive: $std-interface-gray-lighten-1-color !default;

/* Info */
$std-button-color--ghost-info: $std-feedback-info-color !default;
$std-button-color--ghost-info-pressed: map-get(
  $std-list-button-pallete,
  'info-darken-5'
) !default;
$std-button-color--ghost-info-hover: $std-button-color--ghost-info !default;
$std-button-color--ghost-info-focus: $std-button-color--ghost-info-hover !default;
$std-button-color--ghost-info-inactive: $std-interface-gray-lighten-1-color !default;

/* Danger */
$std-button-color--ghost-danger: $std-feedback-danger-color !default;
$std-button-color--ghost-danger-pressed: map-get(
  $std-list-button-pallete,
  'danger-darken-5'
) !default;
$std-button-color--ghost-danger-hover: $std-button-color--ghost-danger !default;
$std-button-color--ghost-danger-focus: $std-button-color--ghost-danger-hover !default;
$std-button-color--ghost-danger-inactive: $std-interface-gray-lighten-1-color !default;

/* Warning */
$std-button-color--ghost-warn: $std-feedback-warn-color !default;
$std-button-color--ghost-warn-pressed: map-get(
  $std-list-button-pallete,
  'warn-darken-5'
) !default;
$std-button-color--ghost-warn-hover: $std-button-color--ghost-warn !default;
$std-button-color--ghost-warn-focus: $std-button-color--ghost-warn-hover !default;
$std-button-color--ghost-warn-inactive: $std-interface-gray-lighten-1-color !default;

/* Success */
$std-button-color--ghost-success: $std-feedback-success-color !default;
$std-button-color--ghost-success-pressed: map-get(
  $std-list-button-pallete,
  'success-darken-5'
) !default;
$std-button-color--ghost-success-hover: $std-button-color--ghost-success !default;
$std-button-color--ghost-success-focus: $std-button-color--ghost-success-hover !default;
$std-button-color--ghost-success-inactive: $std-interface-gray-lighten-1-color !default;

/* Tool */
$std-button-color--ghost-tool: $std-interface-gray-color !default;
$std-button-color--ghost-tool-pressed: $std-interface-gray-darken-1-color !default;
$std-button-color--ghost-tool-hover: $std-interface-gray-darken-1-color !default;
$std-button-color--ghost-tool-focus: $std-interface-gray-darken-1-color !default;
$std-button-color--ghost-tool-inactive: $std-interface-gray-lighten-1-color !default;

$std-ghost-buttons: (
  default: (
    active: $std-button-color--ghost,
    pressed: $std-button-color--ghost-pressed,
    focus: $std-button-color--ghost-focus,
    hover: $std-button-color--ghost-hover,
    inactive: $std-button-color--ghost-inactive,
  ),
  info: (
    active: $std-button-color--ghost-info,
    pressed: $std-button-color--ghost-info-pressed,
    focus: $std-button-color--ghost-info-focus,
    hover: $std-button-color--ghost-info-hover,
    inactive: $std-button-color--ghost-info-inactive,
  ),
  danger: (
    active: $std-button-color--ghost-danger,
    pressed: $std-button-color--ghost-danger-pressed,
    focus: $std-button-color--ghost-danger-focus,
    hover: $std-button-color--ghost-danger-hover,
    inactive: $std-button-color--ghost-danger-inactive,
  ),
  warn: (
    active: $std-button-color--ghost-warn,
    pressed: $std-button-color--ghost-warn-pressed,
    focus: $std-button-color--ghost-warn-focus,
    hover: $std-button-color--ghost-warn-hover,
    inactive: $std-button-color--ghost-warn-inactive,
  ),
  success: (
    active: $std-button-color--ghost-success,
    pressed: $std-button-color--ghost-success-pressed,
    focus: $std-button-color--ghost-success-focus,
    hover: $std-button-color--ghost-success-hover,
    inactive: $std-button-color--ghost-success-inactive,
  ),
  tool: (
    active: $std-button-color--ghost-tool,
    pressed: $std-button-color--ghost-tool-pressed,
    focus: $std-button-color--ghost-tool-focus,
    hover: $std-button-color--ghost-tool-hover,
    inactive: $std-button-color--ghost-tool-inactive,
  ),
);

/**
 * Label Button
**/
$std-button-font-color--label: #68a61d !default;
$std-button-color--label: rgba($std-button-font-color--label, 0.1) !default;
$std-button-color--label-pressed: rgba(
  $std-button-font-color--label,
  0.2
) !default;
$std-button-color--label-hover: rgba(
  $std-button-font-color--label,
  0.15
) !default;
$std-button-color--label-focus: $std-button-color--label-hover !default;
$std-button-color--label-inactive: $std-interface-gray-lighten-1-color !default;

/* Info */
$std-button-font-color--label-info: $std-feedback-info-color !default;
$std-button-color--label-info: rgba($std-feedback-info-color, 0.1) !default;
$std-button-color--label-info-pressed: rgba(
  $std-feedback-info-color,
  0.2
) !default;
$std-button-color--label-info-hover: $std-button-font-color--label-info !default;
$std-button-color--label-info-focus: $std-button-color--label-info-hover !default;
$std-button-color--label-info-inactive: $std-interface-gray-lighten-1-color !default;

/* Danger */
$std-button-font-color--label-danger: $std-feedback-danger-color !default;
$std-button-color--label-danger: rgba($std-feedback-danger-color, 0.1) !default;
$std-button-color--label-danger-pressed: rgba(
  $std-feedback-danger-color,
  0.2
) !default;
$std-button-color--label-danger-hover: $std-button-font-color--label-danger !default;
$std-button-color--label-danger-focus: $std-button-color--label-danger-hover !default;
$std-button-color--label-danger-inactive: $std-interface-gray-lighten-1-color !default;

/* Warning */
$std-button-font-color--label-warn: $std-feedback-warn-color !default;
$std-button-color--label-warn: rgba($std-feedback-warn-color, 0.1) !default;
$std-button-color--label-warn-pressed: rgba(
  $std-feedback-warn-color,
  0.2
) !default;
$std-button-color--label-warn-hover: $std-button-font-color--label-warn !default;
$std-button-color--label-warn-focus: $std-button-color--label-warn-hover !default;
$std-button-color--label-warn-inactive: $std-interface-gray-lighten-1-color !default;

/* Success */
$std-button-font-color--label-success: $std-feedback-success-color !default;
$std-button-color--label-success: rgba(
  $std-feedback-success-color,
  0.1
) !default;
$std-button-color--label-success-pressed: rgba(
  $std-feedback-success-color,
  0.2
) !default;
$std-button-color--label-success-hover: $std-button-font-color--label-success !default;
$std-button-color--label-success-focus: $std-button-color--label-success-hover !default;
$std-button-color--label-success-inactive: $std-interface-gray-lighten-1-color !default;

/* Text button */
$std-button-font-color--label-text: $std-interface-gray-darken-1-color !default;
$std-button-color--label-text: rgba(
  $std-button-font-color--label-text,
  0.1
) !default;
$std-button-color--label-text-pressed: rgba(
  $std-button-font-color--label-text,
  0.2
) !default;
$std-button-color--label-text-hover: rgba(
  $std-button-font-color--label-text,
  0.15
) !default;
$std-button-color--label-text-focus: $std-button-color--label-text-hover !default;
$std-button-color--label-text-inactive: #f6f7f9 !default;
$std-button-font-color--label-text-inactive: $std-interface-gray-lighten-2-color !default;

$std-label-buttons: (
  default: (
    font: $std-button-font-color--label,
    active: $std-button-color--label,
    pressed: $std-button-color--label-pressed,
    focus: $std-button-color--label-focus,
    hover: $std-button-color--label-hover,
    inactive: $std-button-color--label-inactive,
  ),
  info: (
    font: $std-button-font-color--label-info,
    active: $std-button-color--label-info,
    pressed: $std-button-color--label-info-pressed,
    focus: $std-button-color--label-info-focus,
    hover: $std-button-color--label-info-hover,
    inactive: $std-button-color--label-info-inactive,
  ),
  danger: (
    font: $std-button-font-color--label-danger,
    active: $std-button-color--label-danger,
    pressed: $std-button-color--label-danger-pressed,
    focus: $std-button-color--label-danger-focus,
    hover: $std-button-color--label-danger-hover,
    inactive: $std-button-color--label-danger-inactive,
  ),
  warn: (
    font: $std-button-font-color--label-warn,
    active: $std-button-color--label-warn,
    pressed: $std-button-color--label-warn-pressed,
    focus: $std-button-color--label-warn-focus,
    hover: $std-button-color--label-warn-hover,
    inactive: $std-button-color--label-warn-inactive,
  ),
  success: (
    font: $std-button-font-color--label-success,
    active: $std-button-color--label-success,
    pressed: $std-button-color--label-success-pressed,
    focus: $std-button-color--label-success-focus,
    hover: $std-button-color--label-success-hover,
    inactive: $std-button-color--label-success-inactive,
  ),
  text: (
    font: $std-button-font-color--label-text,
    active: $std-button-color--label-text,
    pressed: $std-button-color--label-text-pressed,
    focus: $std-button-color--label-text-focus,
    hover: $std-button-color--label-text-hover,
    inactive: $std-button-color--label-text-inactive,
    font-inactive: $std-button-font-color--label-text-inactive,
  ),
);

/**
 * Icon Button
**/
$std-icon-button-font-size: 18px !default;
$std-icon-button-font-weight: 400 !default;
$std-icon-button-font-color: $std-interface-gray-darken-1-color !default;
$std-icon-button-bg-color: rgba($std-icon-button-font-color, 0.1) !default;
$std-icon-button--pressed: rgba($std-icon-button-font-color, 0.2) !default;
$std-icon-button--hover: rgba($std-icon-button-font-color, 0.15) !default;
$std-icon-button--disabled: #f6f7f9 !default;
$std-icon-button-font-color--disabled: $std-interface-gray-lighten-2-color !default;
$std-icon-button-size: 36px;
$std-icon-button-padding: 6px;

/**
 * List Button
**/
$std-list-button-font-color: $std-interface-gray-darken-1-color !default;
$std-list-button--pressed: $std-primary-one-color !default;
$std-list-button--hover: $std-list-button--pressed !default;

$std-glass-button-flex-gap: 10px !default;
$std-glass-button-padding: 8px 32px !default;
$std-glass-button-border-radius: 3rem !default;
$std-glass-button-background-color: $std-background-white-color !default;
$std-glass-button-font-color: $std-interface-gray-darken-2-color !default;
$std-glass-button-font-color--disabled: $std-interface-gray-lighten-2-color !default;
$std-glass-button-background-color--focus: $std-interface-gray-lighten-2-color !default;
$std-glass-button-background-color--active: $std-interface-gray-lighten-2-color !default;
$std-glass-button-icon-width: 2rem !default;
$std-glass-button-icon-height: 2rem !default;
