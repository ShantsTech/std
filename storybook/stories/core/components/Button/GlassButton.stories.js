import GlassButton from '@ohrm/std/core/components/Button/GlassButton';
import {
  GLASS_ICON_TYPES,
  TYPE_GLASS_VIDEO,
} from '@ohrm/std/core/components/Button/types';

export default {
  title: 'Example/Button/GlassButton',
  component: GlassButton,
  argTypes: {
    icon: {
      control: {type: 'select', options: GLASS_ICON_TYPES},
      defaultValue: TYPE_GLASS_VIDEO,
    },
    style: {control: {type: 'object'}},
    disabled: {control: {type: 'boolean'}},
  },
};

const Template = args => ({
  setup() {
    return {args};
  },
  components: {'std-glass-button': GlassButton},
  template: '<std-glass-button v-bind="args" />',
});

export const Default = Template.bind({});
Default.args = {
  label: 'Share Video',
};
