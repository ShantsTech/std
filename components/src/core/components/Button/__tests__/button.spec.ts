/*
 * This file is part of Shants Tech LLC
 *
 * Copyright (C) 2020 Shants Tech LLC
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see  http://www.gnu.org/licenses
 */

import {mount} from '@vue/test-utils';
import Button from '@ohrm/std/core/components/Button/Button.vue';
import {
  SIZE_LARGE,
  SIZE_MEDIUM,
  SIZE_SMALL,
  TYPE_MAIN,
  TYPE_SECONDARY,
  TYPE_INFO,
  TYPE_DANGER,
  TYPE_WARN,
  TYPE_SUCCESS,
  TYPE_GHOST,
  TYPE_GHOST_INFO,
  TYPE_GHOST_DANGER,
  TYPE_GHOST_WARN,
  TYPE_GHOST_SUCCESS,
  TYPE_LABEL,
  TYPE_LABEL_INFO,
  TYPE_LABEL_DANGER,
  TYPE_LABEL_WARN,
  TYPE_LABEL_SUCCESS,
  TYPE_TEXT,
  TYPE_TOOL,
} from '../types';

describe('Button.vue', () => {
  it('renders STD button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label},
    });
    expect(wrapper.text()).toMatch(label);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD small button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, size: SIZE_SMALL},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD large button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, size: SIZE_LARGE},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD medium button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, size: SIZE_MEDIUM},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD main button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_MAIN},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD secondary button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_SECONDARY},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD solid info button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_INFO},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD solid danger button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_DANGER},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD solid warn button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_WARN},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD solid success button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_SUCCESS},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD ghost button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_GHOST},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD ghost-info button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_GHOST_INFO},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD ghost-danger button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_GHOST_DANGER},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD ghost-warn button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_GHOST_WARN},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD ghost-success button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_GHOST_SUCCESS},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD label button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_LABEL},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD label-info button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_LABEL_INFO},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD label-danger button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_LABEL_DANGER},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD label-warn button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_LABEL_WARN},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD label-success button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_LABEL_SUCCESS},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD text button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_TEXT},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD tool button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, displayType: TYPE_TOOL},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD iconName button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, iconName: 'arrow-left'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD iconRightName button', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, iconRightName: 'arrow-right'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders STD button with custom color', () => {
    const label = 'Button';
    const wrapper = mount(Button, {
      props: {label, style: {backgroundColor: 'palegreen'}},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
