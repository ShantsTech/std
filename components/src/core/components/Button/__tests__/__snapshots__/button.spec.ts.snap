// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button.vue renders STD button 1`] = `
<button type="button" class="std-button std-button--medium std-button--main">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD button with custom color 1`] = `
<button type="button" class="std-button std-button--medium std-button--main" style="background-color: palegreen;">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD ghost button 1`] = `
<button type="button" class="std-button std-button--medium std-button--ghost">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD ghost-danger button 1`] = `
<button type="button" class="std-button std-button--medium std-button--ghost-danger">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD ghost-info button 1`] = `
<button type="button" class="std-button std-button--medium std-button--ghost-info">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD ghost-success button 1`] = `
<button type="button" class="std-button std-button--medium std-button--ghost-success">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD ghost-warn button 1`] = `
<button type="button" class="std-button std-button--medium std-button--ghost-warn">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD iconName button 1`] = `
<button type="button" class="std-button std-button--medium std-button--main"><i class="std-icon bi-arrow-left std-button-icon"></i> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD iconRightName button 1`] = `
<button type="button" class="std-button std-button--medium std-button--main">
  <!--v-if--> Button <i class="std-icon bi-arrow-right std-button-icon"></i>
</button>
`;

exports[`Button.vue should renders STD label button 1`] = `
<button type="button" class="std-button std-button--medium std-button--label">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD label-danger button 1`] = `
<button type="button" class="std-button std-button--medium std-button--label-danger">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD label-info button 1`] = `
<button type="button" class="std-button std-button--medium std-button--label-info">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD label-success button 1`] = `
<button type="button" class="std-button std-button--medium std-button--label-success">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD label-warn button 1`] = `
<button type="button" class="std-button std-button--medium std-button--label-warn">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD large button 1`] = `
<button type="button" class="std-button std-button--large std-button--main">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD main button 1`] = `
<button type="button" class="std-button std-button--medium std-button--main">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD medium button 1`] = `
<button type="button" class="std-button std-button--medium std-button--main">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD secondary button 1`] = `
<button type="button" class="std-button std-button--medium std-button--secondary">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD small button 1`] = `
<button type="button" class="std-button std-button--small std-button--main">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD solid danger button 1`] = `
<button type="button" class="std-button std-button--medium std-button--danger">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD solid info button 1`] = `
<button type="button" class="std-button std-button--medium std-button--info">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD solid success button 1`] = `
<button type="button" class="std-button std-button--medium std-button--success">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD solid warn button 1`] = `
<button type="button" class="std-button std-button--medium std-button--warn">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD text button 1`] = `
<button type="button" class="std-button std-button--medium std-button--text">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;

exports[`Button.vue should renders STD tool button 1`] = `
<button type="button" class="std-button std-button--medium std-button--tool">
  <!--v-if--> Button
  <!--v-if-->
</button>
`;
